File: ABC178/C_Ubiquity/submission1.scala
// AC

object Main extends App {
  val m = 1000000007;
  def pow(a: Long, b: Int): Long = {
    var x: Long = 1;
    var a2: Long = a;
    var b2: Int = b;
    while (b2 > 0) {
      if ((b2 & 1) != 0) {
        x = x * a2 % m;
      }
      a2 = a2 * a2 % m;
      b2 = b2 >> 1;
    }
    x;
  }
  val sc = new java.util.Scanner(System.in);
  val n = sc.nextInt();
  val a1 = pow(10, n); // all
  val a2 = pow(8, n);  // 0も9も存在しない
  val a3 = (pow(9, n) + m - a2) % m;  // 0は存在するけど9は存在しない
  val answer = (a1 + 3L * m - 2L * a3 - a2) % m;
  println(answer);
}

File: ABC179/D_Leaping_Tak/submission1.java
// TLE

import java.util.Scanner;

class Main {
  private static int[] table;
  private static int m = 998244353;
  private static int[][] lrs;

  private static int fetch(int x) {
    if (table[x] > 0) {
      return table[x];
    }
    int sum = 0;
    for (int i = 0; i < lrs.length; i++) {
      int s = lrs[i][0];
      int e = lrs[i][1];
      for (int j = s; j <= e; j++) {
        int x2 = x - j;
        if (x2 <= 0) break;
        sum += fetch(x2);
        sum = sum % m;
      }
    }
    table[x] = sum;
    return sum;
  }

  public static void main(String[] args) {
    var sc = new Scanner(System.in);
    int n = sc.nextInt();
    int k = sc.nextInt();
    lrs = new int[k][2];
    for (int i = 0; i < k; i++) {
      lrs[i][0] = sc.nextInt();
      lrs[i][1] = sc.nextInt();
    }
    table = new int[n + 1];
    table[1] = 1;

    for (int i = 2; i <= n; i++) {
      fetch(i);
    }
    var answer = fetch(n);
    System.out.println(answer);
  }
}
